sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/ActionDescriptorFactory.scala

```diff
@@ -47,12 +47,34 @@ import kalix.javasdk.impl.reflection.NameGenerator
 import kalix.javasdk.impl.reflection.ReflectionUtils
 import kalix.javasdk.impl.reflection.RestServiceIntrospector
 import kalix.javasdk.impl.reflection.SubscriptionServiceMethod
-import java.lang.reflect.Method
 
+import java.lang.reflect.Method
 import kalix.javasdk.impl.ComponentDescriptorFactory.mergeServiceOptions
+import kalix.javasdk.impl.Reflect.Syntax.AnnotatedElementOps
+import kalix.javasdk.impl.Reflect.Syntax.MethodOps
+import kalix.TriggerOptions
+import kalix.javasdk.annotations.Trigger
+
+import scala.PartialFunction.condOpt
 
 private[impl] object ActionDescriptorFactory extends ComponentDescriptorFactory {
 
+  private def hasTriggerMethodOptions(javaMethod: Method): Boolean = {
+    javaMethod.isPublic && javaMethod.hasAnnotation[Trigger.OnStartup] // this is the only event available at the moment
+  }
+
+  private def triggerOptions(javaMethod: Method): Option[TriggerOptions] = {
+    condOpt(hasTriggerMethodOptions(javaMethod)) { case true =>
```

Comment:
Here we can use Option.when instead.

```suggestion
    Option.when(hasTriggerMethodOptions(javaMethod)) {
```